service/src/main/java/bio/terra/drshub/services/DrsProviderService.java

```diff
@@ -37,31 +40,46 @@ public record DrsProviderService(DrsHubConfig drsHubConfig) {
    * that if the host part of the URI is of the form dg.[0-9a-z-]+ or drs.[0-9a-z-]+ then it is a
    * Compact Identifier.
    *
+   * <p>Hostname ID: drs://hostname/id
+   *
+   * <p>https://drs.example.org/ga4gh/drs/v1/objects/314159
+   *
+   * <p>Compact ID: drs://prefix:accession
+   *
+   * <p>drs://dg.anv0:f51fc329-b09e-4e16-b1a9-2f60ebc428ab
+   *
    * <p>If you update *any* of the below be sure to link to the supporting docs and update the
    * comments above!
    */
   public UriComponents getUriComponents(String drsUri) {
 
     var compactIdMatch = compactIdRegex.matcher(drsUri);
-    final String dnsHost;
+    final String hostString;
     final String strippedPath;
     var hostNameMatch = hostNameRegex.matcher(drsUri);
+    var schemeMatch = schemeRegex.matcher(drsUri);
 
+    // TODO: If ID is compact we need to url encode any slashes
```

Comment:
Ill make a ticket for this tuesday, see this code in martha https://github.com/broadinstitute/martha/blob/84b001264daae5977218ab5b80bb03b37e9b9baf/martha/martha_v3.js#L162